src/snapshot/tosnapshot.js

```diff
@@ -0,0 +1,74 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Plotly = require('../plotly');
+var Lib = require('../lib');
+
+/**
+ * @param {object} gd figure Object
+ * @param {object} opts option object
+ * @param opts.format 'jpeg' | 'png' | 'webp' | 'svg'
+ */
+function toSnapshot(gd, opts) {
```

Comment:
we should probably expose a *download image* functionality too through the `Plotly` object.

Maybe:

```js 
Plotly.plot('graph', data, layout)
 .then(Plotly.downloadImage({ /* e.g.: format: 'png', width: 700, height: 500 */ });
```

@timelyportfolio thoughts?